lib/max_subarray.rb

```diff
@@ -2,7 +2,25 @@
 # Time Complexity: ?
 # Space Complexity: ?
 def max_sub_array(nums)
```

Comment:
This works, but no guesses on time/space complexity?

Reply:
Whoops, totally missed those!  I think they should both be O(n).